spec/ParseLiveQueryServer.spec.js

```diff
@@ -1348,6 +1606,29 @@ describe('ParseLiveQueryServer', function() {
       });
   });
 
+  it('should properly pull auth from cache', () => {
+    const parseLiveQueryServer = new ParseLiveQueryServer({});
+    const promise = parseLiveQueryServer.getAuthForSessionToken('sessionToken');
+    const secondPromise = parseLiveQueryServer.getAuthForSessionToken(
+      'sessionToken'
+    );
+    // should be in the cache
+    expect(parseLiveQueryServer.authCache.get('sessionToken')).toBe(promise);
+    // should be the same promise returned
+    expect(promise).toBe(secondPromise);
+    // the auth should be called only once
+    expect(auth.getAuthForSessionToken.calls.count()).toBe(1);
+  });
+
+  it('should delete from cache throwing auth calls', async () => {
```

Comment:
i'm not sure what you mean by throwing here.  I know that your sessiontoken 'pleaseThrow' returns a rejected promise, but no errors are being thrown, right?  I'm confused by the choice of words maybe?

Reply:
`parseLiveQueryServer.getAuthForSessionToken('pleaseThrow');` provides a session token that yeilds an error, the method tests that the error is not propagated and properly invalidated the cache.